generic-server/src/plugin/container.rs

```diff
@@ -185,87 +192,143 @@ mod tests {
 
     #[test]
     fn test_loading() {
+        // Mock plugins for testing
+        let plugins: Arc<Vec<Arc<Mutex<dyn Plugin>>>> = Arc::new(vec![
+            Arc::new(Mutex::new(FirstPlugin {})),
+            Arc::new(Mutex::new(SecondPlugin {})),
+        ]);
+
+        // Initialize PluginContainer with the mock plugins
         let mut container = PluginContainer {
             loaded: false,
             collected_routes: vec![],
-            plugins: &vec![Box::new(FirstPlugin {}), Box::new(SecondPlugin {})],
+            plugins: Arc::clone(&plugins),
         };
 
+        // Test loading plugins
         assert!(container.load().is_ok());
         assert!(container.routes().is_ok());
 
+        // Verify find_plugin method
         assert!(container.find_plugin("first").is_some());
         assert!(container.find_plugin("second").is_some());
         assert!(container.find_plugin("non-existent").is_none());
 
-        // The actual routes collected are actually hard to test
-        // given that axum::Router seems not to provide public
-        // directives to inquire internal state.
-        // See: https://github.com/tokio-rs/axum/discussions/860
+        // Verify collected routes
         assert_eq!(container.collected_routes.len(), 2);
     }
 
     #[test]
     fn test_double_loading() {
+        // Mock plugins for testing
+        let plugins: Arc<Vec<Arc<Mutex<dyn Plugin >>>> = Arc::new(vec![
+            Arc::new(Mutex::new(FirstPlugin {})),
+            Arc::new(Mutex::new(SecondPlugin {})),
+        ]);
+    
+        // Initialize PluginContainer with the mock plugins
         let mut container = PluginContainer {
             loaded: false,
             collected_routes: vec![],
-            plugins: &vec![Box::new(FirstPlugin {}), Box::new(SecondPlugin {})],
+            plugins: Arc::clone(&plugins),
         };
-
-        assert!(container.load().is_ok());
+    
+        // Test loading plugins twice
         assert!(container.load().is_ok());
-
+        assert!(container.load().is_ok()); // Load again, should succeed without errors
+    
+        // Verify collected routes
         assert_eq!(container.collected_routes.len(), 2);
     }
-
+    
     #[test]
     fn test_loading_with_duplicates() {
+        // Mock plugins for testing with duplicate name "second"
```

Comment:
Make sure your changes do not alter the narrative of this test.

Reply:
Check the original test. It asserts that the logic raises a DuplicateEntry error. Does your changes do that?
![image](https://github.com/adorsys/didcomm-mediator-rs/assets/30264663/e680b595-e7f7-4aa3-ba79-8515cb647280)

Reply:
yes sir when you try to  load plugins with duplicate names

Reply:
An assert is different than printing messages. Use an assert on the result, instead of the match.

```rust
assert_eq!(result.unwrap_err(), PluginContainerError::DuplicateEntry);
```

Reply:
okay sir thanks i have correct